File: CodeFreak1/ViewModel/ProblemCompleteViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CodeFreak1.ViewModel
{
    public class ProblemCompleteViewModel: RequestStatus
    {
        public ProblemViewModel Problem { get; set; }
        public ProblemTypeViewModel ProblemType { get; set; }
        public DifficultyViewModel Difficulty { get; set; }
    }
}

File: CodeFreak1/ClientApp/src/app/problem/dtos/problem-complete-view-model.ts
import { ProblemViewModel } from "./problem-view-model";
import { ProblemTypeViewModel } from "../../problem-type/dtos/problem-type-view-model";
import { DifficultyViewModel } from "../../difficulty/dtos/difficulty-view-model";
import { RequestStatus } from "../../request-status";

export class ProblemCompleteViewModel extends RequestStatus {
  public Problem: ProblemViewModel;
  public ProblemType: ProblemTypeViewModel;
  public Difficulty: DifficultyViewModel;

  constructor() {
    super();
    this.Difficulty = new DifficultyViewModel();
    this.Problem = new ProblemViewModel();
    this.ProblemType = new ProblemTypeViewModel();
  }
}
